modules/util/sec_to_time.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 Gitea. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package util
+
+import "fmt"
+
+// SecToTime converts an amount of seconds to a human-readable string (example: 66s -> 1min 6s)
+func SecToTime(duration int64) string {
+	seconds := duration % 60
+	minutes := (duration / (60)) % 60
+	hours := duration / (60 * 60) % 24
+	days := duration / (60 * 60) / 24
+
+	var hrs string
+
+	if days > 0 {
+		hrs = fmt.Sprintf("%dd", days)
+	}
+	if hours > 0 {
+		if days == 0 {
```

Comment:
I think these can be switched to `hrs == ""` or `len(hrs) == 0`, better to have only 1 condtion instead of checking all of them.